editor/src/core/model/element-template-utils.ts

```diff
@@ -511,6 +512,65 @@ export function insertChildAndDetails(
 }
 
 export function insertJSXElementChild(
+  targetParent: InsertionPath,
+  elementToInsert: JSXElementChild,
+  components: Array<UtopiaJSXComponent>,
+  indexPosition: IndexPosition | null,
+): InsertChildAndDetails {
+  const makeE = () => {
+    // TODO delete me
+    throw new Error('Should not attempt to create empty elements.')
+  }
```

Comment:
could this be deleted, and instead an explicit check be made whether `indexPosition` points to an index that won't make it necessary to create empty elements?

Reply:
fixed in 32972823e